Dockerfile.builder

```diff
@@ -14,7 +14,8 @@ WORKDIR /go/src/github.com/cilium/cilium
 ENV GOROOT /usr/local/go
 ENV GOPATH /go
 ENV PATH "$GOROOT/bin:$GOPATH/bin:$PATH"
-ENV GO_VERSION 1.14.1
+ENV GO_VERSION 1.14
```

Comment:
Why change `GO_VERSION` here? Was this regenerated somehow using an old `GO_VERSION` file. I think we should stay on 1.14.1.